src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -441,40 +457,31 @@ private static void RemoveDocument(ImmutableArray<IIncrementalAnalyzer> analyzer
                         }
                     }
 
-                    private void ResetLogAggregatorIfNeeded(Solution currentSolution)
-                    {
-                        if (currentSolution == null || _processingSolution == null ||
-                            currentSolution.Id == _processingSolution.Id)
-                        {
-                            return;
-                        }
-
-                        SolutionCrawlerLogger.LogIncrementalAnalyzerProcessorStatistics(
-                            this.Processor._registration.CorrelationId, _processingSolution, this.Processor._logAggregator, this.Analyzers);
-
-                        this.Processor.ResetLogAggregator();
-                    }
+                   // this is only used in ResetState to find out solution has changed
+                   // and reset some states
+                    private Solution _lastSolution = null;
```

Comment:
nit: Move private field to the top of type definition

Reply:
can this explanation be expanded on.  IT tells me 'what', but not 'why'.

Reply:
can we document threading expectations here?  Is this field used across multiple threads?

Reply:
that field is supposed to be used only in ResetState, that's why I moved here, but sure, I can move it back up to the top.

sure, I can add why part as well.